builder/dockerfile/internals.go

```diff
@@ -489,9 +489,13 @@ func hostConfigFromOptions(options *types.ImageBuildOptions, isWCOW bool) *conta
 	// For WCOW, the default of 20GB hard-coded in the platform
 	// is too small for builder scenarios where many users are
 	// using RUN statements to install large amounts of data.
-	// Use 127GB as that's the default size of a VHD in Hyper-V.
-	if isWCOW {
-		hc.StorageOpt = make(map[string]string)
+	// If no custom size is set, use 127GB for containers that
+	// are started during build, as that's the default size of
+	// a VHD in Hyper-V.
+	if isWCOW && hc.StorageOpt["size"] == "" {
```

Comment:
While you're at it, you might want to look at the definition of `isWCOW` on line 451. Currently, it's true if you run `docker build . --platform windows` but *false* if you run `docker build .`. I was just now looking into whether it should be inverted to be
```
isWCOW := runtime.GOOS == "windows" && (b.platform == nil || b.platform.OS == "windows")
```

Reply:
Oh! Comments crossed with the last comment I left on https://github.com/moby/moby/issues/37352#issuecomment-722315438

> Ah, interesting; sounds like more fun because of the LCOW paths (I guess it doesn't set WCOW if it doesn't know if the image is gonna be Linux or Windows?) haven't checked all paths
>
> Should we remove the isWCOW there? It would be a somewhat closer match to "linux" (where by default size is only restricted by physical disk space)and replace with `if runtime.GOOS == "windows"`

WDYT?

Reply:
LCOW has its own option, `lcow.sandboxsize`, so `size` would be ignored by LCOW. We can't validate that because the config is shared between WCOW and LCOW. -_-

I would _expect_ that if `runtime.GOOS == "windows"`, then LCOW would only happen if `b.platform.OS == "linux"`, which I assume is what this check was _intended_ to do.

Or perhaps at the time, `b.platform` would not reasonably be `nil`, but now we have things that check `platform == nil` and fallback to `runtime.GOOS`, e.g., `createCopyInstruction` in _builder/dockerfile/copy.go_.

Anyway, given that, a simple check for `runtime.GOOS` would be okay, since we know the only graphdrivers that run on Windows are lcow and windowsfilter.

Reply:
LCOW is also a 20GB default, it might make sense that we set both `size=` and `lcow.sandboxsize=` to 127GB here.